### PR TITLE
Add initial ingester end-to-end tests

### PR DESCRIPTION
**Pull request by juliusv**

### CODE REVIEW

File: ingester/ingester_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+		ss.Values = append(ss.Values, samples...)
+	}
+
+	matrix := make(model.Matrix, 0, len(sampleStreams))
+	for _, ss := range sampleStreams {
+		matrix = append(matrix, ss)
+	}
+	return matrix
+}
```
Comment (tomwilkie) [inline ingester/ingester_test.go]: Could this be done with the code in https://github.com/weaveworks/cortex/blob/master/querier/querier.go#L38?

Comment (juliusv) [inline ingester/ingester_test.go]: Yeah, that's where I copied it from :) Was thinking about factoring that part out, but then I wasn't sure where to put it. Maybe just a `ChunksToMatrix([]Chunk) model.Matrix` function in the `chunk` package?

**Comment by tomwilkie** [comment]
Yeah sounds like a plan.

File: ingester/ingester_test.go (hunk 1 above)
Comment (juliusv) [inline ingester/ingester_test.go]: This ends up being a larger refactoring due to transitive dependencies and entanglements. I've got the result already locally, but I'd want to test it a bit more to make sure I didn't break any of the chunk sorting in other places which were touched as well. How about merging this one first and I'll do the refactoring as a followup.

**Comment by tomwilkie** [comment]
SGTM

**Pull request merged**